airbyte-cdk/python/airbyte_cdk/sources/declarative/yaml_declarative_source.py

```diff
@@ -42,9 +48,4 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
     def _read_and_parse_yaml_file(self, path_to_yaml_file):
         with open(path_to_yaml_file, "r") as f:
             config_content = f.read()
-            parsed_config = YamlParser().parse(config_content)
```

Comment:
how does moving this help? the `debug` level was already set above it no?

Reply:
Actually when we construct the declarative source in `_read_and_parse_yaml_file()` in source.py (here's how its implemented for [source-sendgrid](https://github.com/airbytehq/airbyte/pull/15344/files#diff-fe556ab8b8fc837d427787eb85eea6b94560a58306d032e01f9fe4ac02439f25)), this is done before we start `entrypoint.py.launch()`. [main.py](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-sendgrid/main.py) for sengrid

So we don't actually know yet if the user specified `--debug` since that gets interpreted in launch. When we move it to `streams()`this gets called from `entrypoint.py` after log level is set.

Reply:
and the above debug level should be removed since its effectively a no-op, since it gets reassigned in entrypoint.

Reply:
We should probably add a todo to resolve this but nbd for this pr

Reply:
I removed the debug level in the above code, so I don't think we'd need any follow ups. As it stands now the flow of the read command is:

1. parse yaml
2. save it to `self._source_config`
3. parse command line args in entrypoint
4. assign logger level to DEBUG or INFO based on presence of `--debug`
5. handle `read`
6. emit debug message with parsed YAML if on
7. process sync